test/README.md

```diff
@@ -26,12 +26,48 @@ Make sure you have valid credentials for accessing GCP:
 gcloud auth login
 ```
 
+Set the project property in the core section by running:
+```
+gcloud config set project PROJECT_ID
+```
+
+Installed jq by running.
+```
+sudo apt-get install jq
+```
+
 Make sure you are at the root directory of the repository.
 
 ```
 cd gke-networking-recipes
 ```
 
+
+First, set ZONE and SUBNET_REGION environment variable, then run ./test/generate_config.sh to generate the testing config.
+```
+export ZONE=zone
+export SUBNET_REGION=subnet_region
+
+./test/generate_config.sh
+```
+
+To run ingress-https test, you need a special project($DNS_PROJECT) that has configured DNS domain, and make sure the current service account have role/dns_admin access to that project. 
+Create a public DNS zone(DNS_ZONE). Then set the environment variable for DNS_PROJECT, HTTPS_RECORD1, HTTPS_RECORD2, and DNS_ZONE before running generate_env.sh. Make sure these records that uses the DNS name of the created DNS zone as suffix.
+To run ingress-iap and ingress_asm test, you need to have a support email that follows the requirement described in https://cloud.google.com/iap/docs/programmatic-oauth-clients and export it as SUPPORT_EMAIL.
+For ingress_iap test, you also need an IAP_RECORD that uses the DNS zone as suffix.
+```
+export ZONE=zone \
+export SUBNET_REGION=subnet_region \
+export DNS_PROJECT=dns_project \
+export DNS_ZONE=dns_zone \
+export HTTPS_RECORD1=https_record1 \
```

Comment:
what is HTTPS_RECORD1,2 mean?

Reply:
They are two HTTPS records needed to replace the placeholder in ingress-https yaml.